packages/messaging/ios/RNFBMessaging/RNFBMessaging+UNUserNotificationCenter.m

```diff
@@ -88,7 +88,18 @@ - (void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNoti
   NSDictionary *remoteNotification = response.notification.request.content.userInfo;
   if (remoteNotification[@"gcm.message_id"]) {
     NSDictionary *notificationDict = [RNFBMessagingSerializer remoteMessageUserInfoToDict:remoteNotification];
-    [[RNFBRCTEventEmitter shared] sendEventWithName:@"messaging_notification_opened" body:notificationDict];
+
+    // create a merged dictionary that will hold the serialized userInfo data, as well as any actionIdentifier
+    NSMutableDictionary * mergedDict = [NSMutableDictionary dictionary];
+
+    // copy the values of notificationDict into mergedDict
+    [mergedDict addEntriesFromDictionary:notificationDict];
+
+    // if there is an actionIdentifier on this notification, also add that in.
+    [mergedDict setObject:response.actionIdentifier forKey:@"actionIdentifier"];
+
```

Comment:
First, thanks very much for doing this! Our team literally just finalized Q3 planning which included investigating just this type of PR!

As an aside, would you also consider including the [TextInputNotificationResponse](https://developer.apple.com/documentation/usernotifications/untextinputnotificationresponse) for completeness? This is for notifications that are configured to receive user text input e.g. a reply message to an incoming notification.

Something to the effect of:
```
  if ([response isKindOfClass:UNTextInputNotificationResponse.class]) {
    [mergedDict setValue:[(UNTextInputNotificationResponse *)response userText] forKey:@"userInput"];
  }
```

If not, I totally understand. We can submit a supplemental PR (or consider using patch-package, etc)

Thanks again!

Reply:
...along with the same wish for a change to the index.d.ts for that key as well :-), but I'd merge that yes.
@TNicholson11 - any thoughts on an android implementation for parity? or is it not needed

Reply:
Happy to be wrong on this one (since it would be very helpful for our product) but as best I can tell, you cannot set Android actions on a remote message therefore there would be no response to pass along. On iOS, the notification categories are pre-registered so the iOS does it's magic on the incoming remote message. On Android, that is seemingly only supported for local notifications via Notifee (for example).

The response from a local notification on Android is included in the clickAction property of the [android payload](https://rnfirebase.io/reference/messaging/notification#android)

Long way of saying, to the best of my knowledge there isn't an Android solution required at this time :)

Reply:
Yes I can back that up @TNicholon11, Android can only present action buttons via the silent remote / rich local flow. Which is bad news for us (because we need Quick Actions on our Android app), but good news for this PR 😅

I'll do the type definition for both `actionIdentifier` and `userText` and push here, if I have trouble getting a prototype of text input to work I'll post here

Reply:
On the plus side, Android data-only delivery is reliable in my experience? Well, as long as the vendor hasn't destroyed it (https://dontkillmyapp.com) but in general I have no issues on Android.